maintenance/troubleshoot/commands.md

```diff
@@ -0,0 +1,499 @@
+calico-node-9t8s7
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-9cjhw
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-cb7ff
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-qoxvw
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-wm5m2
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+```
+
+#### View pod info of pod that is not running
+
+```bash
+kubectl describe pods `<pod_name>`  -n `<namespace> `
+```
+
+```bash
+kubectl describe pods busybox -n default
+```
+
+```
+Events:
+  Type    Reason     Age   From                   Message
+  ----    ------     ----  ----                   -------
+  Normal  Scheduled  21s   default-scheduler      Successfully assigned default/busybox to ip-10-0-0-11
+  Normal  Pulling    20s   kubelet, ip-10-0-0-11  Pulling image "busybox"
+  Normal  Pulled     19s   kubelet, ip-10-0-0-11  Successfully pulled image "busybox"
+  Normal  Created    19s   kubelet, ip-10-0-0-11  Created container busybox
+  Normal  Started    18s   kubelet, ip-10-0-0-11  Started container busybox
+```
+
+#### View logs of a pod that is not running
+
+```bash
+kubectl logs `<pod_name>`  -n `<namespace>`
+```
+
+```bash
+kubectl logs busybox -n default
+```
+
+#### View kernel logs
+
+```bash
+journalctl -u kubelet
+```
+
+### Routing
+
+#### Verify routing table on the node
+
+Look for common issues like TCP port XXX is not open between instances.
+
+```bash
+ip route
+```
+
+```
+default via 10.0.0.1 dev eth0 proto dhcp src 10.0.0.10 metric 100 
+10.0.0.0/24 dev eth0 proto kernel scope link src 10.0.0.10 
+10.0.0.1 dev eth0 proto dhcp scope link src 10.0.0.10 metric 100 
+10.48.66.128/26 via 10.0.0.12 dev eth0 proto 80 onlink 
+10.48.231.0/26 via 10.0.0.11 dev eth0 proto 80 onlink 
+172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown
+```
+
+#### Verify BGP peer status
+
+```bash
+sudo calicoctl node status
+```
+
+```
+Calico process is running.
+
+IPv4 BGP status
++--------------+-------------------+-------+------------+-------------+
+| PEER ADDRESS |     PEER TYPE     | STATE |   SINCE    |    INFO     |
++--------------+-------------------+-------+------------+-------------+
+| 10.0.0.12    | node-to-node mesh | up    | 2020-11-25 | Established |
+| 10.0.0.11    | node-to-node mesh | up    | 2020-11-25 | Established |
++--------------+-------------------+-------+------------+-------------+
+```
+
+#### Verify overlay configuration
+
+```bash
+kubectl get ippools default-ipv4-ippool -o yaml
+```
+
+```yaml
+...
+spec:
+  ipipMode: Always
+  vxlanMode: Never
+...
+```
+
+#### Verify bgp learned routes
```

Comment:
Same here - the new CalicoNodeStatus API provides a much better way to do this.

Reply:
Added to ticket.